pkg/api/controller.go

```diff
@@ -3906,8 +3906,89 @@ func (c *Controller) PostStatsEvents(w http.ResponseWriter, r *http.Request, bod
 
 func (c *Controller) OtfDiff(w http.ResponseWriter, r *http.Request, repository string, leftRef string, rightRef string, params OtfDiffParams) {
 	ctx := r.Context()
+	user, err := auth.GetUser(ctx)
+	if err != nil {
+		writeError(w, r, http.StatusUnauthorized, ErrAuthenticatingRequest)
+		return
+	}
 	c.LogAction(ctx, fmt.Sprintf("table_format_%s_diff\n", params.Type), r, repository, rightRef, leftRef)
-	writeResponse(w, r, http.StatusOK, nil)
+	credentials, _, err := c.Auth.ListUserCredentials(ctx, user.Username, &model.PaginationParams{
+		Prefix: "",
+		After:  "",
+		Amount: 1,
+	})
+	if c.handleAPIError(ctx, w, r, err) {
+		return
+	}
+	if len(credentials) == 0 {
```

Comment:
there's a lot to do with regard to all of the authentication details and it will be made on later PRs.
In the meantime I don't think there's a need to comment or remove this condition